docs/development/component-checklist.md

```diff
@@ -0,0 +1,146 @@
+# Checklist For Adding New Components
+
+Adding new components which run in garden, seed or shoot cluster is theoretically quite simple - we just need a `Deployment` (or similar other workload resource), the respective container image and maybe a bit of configuration.
+In practice however, there are a couple of things to keep in mind in order to make the deployment production-ready.
+This document provides a checklist for them which you can walk through. 
+
+## General
+
+1. **Avoid usage of Helm charts** ([example](https://github.com/gardener/gardener/tree/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/metricsserver))
+
+   Nowadays, we use [Golang components](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/interfaces.go) instead of Helm charts for deploying components to a cluster.
+   There are a few exceptions (e.g., [Istio](https://github.com/gardener/gardener/tree/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/istio)) still using charts, however the default should be using a Golang-based implementation.
+
+2. **Choose the proper deployment way** ([example 1](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/kubescheduler/kube_scheduler.go#L210-L225), [example 2](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/kubescheduler/kube_scheduler.go#L442-L484), [example 3](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/kubestatemetrics/kube_state_metrics.go#L116))
```

Comment:
Not sure whether I can come up with some artificial names right now 😅 Any suggestions? The text in this section should explain the three cases (direct application with client, GRM only, or mixed scenario with the utility functions). Isn't this good enough/what do you miss?

Reply:
Naming the examples accordingly would already clarify it.
`direct application with client`, `gardener-resource-manager` and `mixed scenario` instead of `example [1|2|3]`

Reply:
done w/ 8fc37e64f